core/src/main/scala/com/microsoft/azure/synapse/ml/param/JsonEncodableParam.scala

```diff
@@ -179,10 +194,12 @@ class CognitiveServiceStructParam[T: TypeTag](parent: Params,
                                               isValid: T => Boolean = (_: T) => true)
                                              (@transient implicit val dataFormat: JsonFormat[T])
   extends JsonEncodableParam[T](parent, name, doc, isValid)
-    with PythonWrappableParam[T] with WrappableParam[T] {
+    with PythonWrappableParam[T] with RWrappableParam[T] with WrappableParam[T] {
```

Comment:
Consider refactoring this class heirarchy so that WrappableParam methods get pushed into DotNetWrappableParam (because for some reason there are only dotnet methods in WrappableParam API). Then we can make WrappableParam just combine PythonWrappableParam, RWrappableParam, and DotNetWrappableParam

Reply:
Can we leave this for a different PR? It seems tangential to this one, and when I tried this, it caused lots of errors since pyValue and rValue become undefined in many subclasses.

Reply:
If you want i can try my hand at this. This seems like its a bad smell and we should try to take it out now, especially if the code is all tangled up from it

Reply:
Sure, if you want to try, have at it.